www_src/lib/platform.js

```diff
@@ -0,0 +1,171 @@
+/**
+ * Device agnostic platform interface.
+ *
+ * @package core
+ * @author Andrew Sliwinski <a@mozillafoundation.org>
+ */
+
+var lru = require('lru-cache');
+
+/**
+ * Constructor
+ */
+function Platform () {
+  // Check to see if window.Platform exists. If it does, return it.
+  if (window.Platform) {
+    return window.Platform;
+  }
+
+  // Init storage objects
+  this.sharedStorage = window.localStorage;
+  this.memStorage = lru(50);
+  this.sessionKey = 'WEBMAKER_SESSION';
+  this.cacheKey = '::' + window.pathname;
+}
+
+// -----------------------------------------------------------------------------
+// Persistent storage
+// -----------------------------------------------------------------------------
+
+Platform.prototype.getSharedPreferences = function(key, global) {
+  if (!global) {
+    key += this.cacheKey;
+  }
+  return this.sharedStorage.getItem(key);
+};
+
+Platform.prototype.setSharedPreferences = function(key, value, global) {
+  if (!global) {
+    key += this.cacheKey;
+  }
+  return this.sharedStorage.setItem(key, value);
+};
+
+Platform.prototype.resetSharedPreferences = function() {
+  return this.sharedStorage.clear();
+};
+
+// -----------------------------------------------------------------------------
+// Session storage
+// -----------------------------------------------------------------------------
+
+Platform.prototype.getUserSession = function() {
```

Comment:
`getUserSession` uses the shared preferences API to get and set the user session using `this.sessionKey`.